src/Features/Core/Portable/IntroduceVariable/AbstractIntroduceParameterService.cs

```diff
@@ -0,0 +1,829 @@
+        /// 
+        /// public void M(int x, int y)
+        /// {
+        ///     int f = x * y;
+        /// }
+        /// </summary>
+        private static async Task GenerateNewMethodAsync(Document document, TExpressionSyntax expression,
+            IMethodSymbol methodSymbol, SyntaxNode containingMethod, string newMethodIdentifier, SyntaxEditor editor, CancellationToken cancellationToken)
+        {
+            // Remove trailing trivia because it adds spaces to the beginning of the following statement.
+            var newStatements = SpecializedCollections.SingletonEnumerable(editor.Generator.ReturnStatement(expression.WithoutTrailingTrivia()));
+
+            var semanticModel = await document.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+            var typeSymbol = semanticModel.GetTypeInfo(expression, cancellationToken).ConvertedType ?? semanticModel.Compilation.ObjectType;
+            var newMethod = CodeGenerationSymbolFactory.CreateMethodSymbol(methodSymbol, name: newMethodIdentifier, returnType: typeSymbol);
+
+            var newMethodDeclaration = editor.Generator.MethodDeclaration(newMethod, newStatements);
+            editor.InsertBefore(containingMethod, newMethodDeclaration);
+        }
+
+        /// <summary>
+        /// Generates a method declaration containing a call to the method that introduced the parameter.
+        /// Example:
+        /// public void M(int x, int y)
+        /// {
+        ///     int f = x * y; // highlight this expression
+        /// }
+        /// 
+        /// ---------------------------------------------------->
+        /// 
+        /// public void M(int x, int y)
+        /// {
+        ///     M(x, y, x * y);
+        /// }
+        /// 
+        /// public void M(int x, int y)
+        /// {
+        ///     int f = x * y;
+        /// }
+        /// </summary>
+        private static void GenerateNewMethodOverload(TExpressionSyntax expression,
+            IMethodSymbol methodSymbol, SyntaxNode containingMethod, SyntaxEditor editor)
+        {
+            var generator = editor.Generator;
+            var arguments = generator.CreateArguments(methodSymbol.Parameters);
+
+            // Remove trailing trivia because it adds spaces to the beginning of the following statement.
+            arguments = arguments.Add(generator.Argument(expression.WithoutTrailingTrivia()));
+            var memberName = methodSymbol.IsGenericMethod
+                ? generator.GenericName(methodSymbol.Name, methodSymbol.TypeArguments)
+                : generator.IdentifierName(methodSymbol.Name);
+
+            var invocation = generator.InvocationExpression(memberName, arguments);
+            var newStatements = SpecializedCollections.SingletonEnumerable(invocation);
+
+            if (!methodSymbol.ReturnsVoid)
+            {
+                newStatements = SpecializedCollections.SingletonEnumerable(generator.ReturnStatement(invocation));
+            }
```

Comment:
we should either make an expression statement, or a return statement.  i don't get how we can add an invocatino as the body of a void method.  can you add test?

Reply:
I do have a test for it, it's called "TestSimpleExpressionWithSingleMethodCallOverload".